index.bs

```diff
@@ -1004,33 +1049,95 @@ To <dfn>trigger attribution</dfn> given an [=attribution trigger=] |trigger| run
+1. If |report|'s [=aggregatable report/source debug key=] is not null and |report|'s
+    [=aggregatable report/trigger debug key=] is not null, [=queue a task=] to
+    [=attempt to deliver a debug report=] with |report|.
+1. Return <strong>attributed</strong>.
+
+<h3 algorithm id="triggering-attribution">Triggering attribution</h3>
+
+To <dfn>trigger attribution</dfn> given an [=attribution trigger=] |trigger| run the following steps:
+
+1. Let |attributionDestination| be |trigger|'s [=attribution trigger/attribution destination=].
+1. Let |matchingSources| be all entries in the [=attribution source cache=] where all of the following are true:
+     * entry's [=attribution source/attribution destination=] and |attributionDestination| are equal.
+     * entry's [=attribution source/reporting endpoint=] and |trigger|'s [=attribution trigger/reporting endpoint=] are equal.
+     * entry's [=attribution source/expiry time=] is greater than the current time.
+1. If |matchingSources| is empty, return.
+1. Set |matchingSources| to the result of [=list/sort in descending order|sorting=] |matchingSources|
+    in descending order, with |a| being less than |b| if any of the following are true:
+      * |a|'s [=attribution source/priority=] is less than |b|'s [=attribution source/priority=].
+      * |a|'s [=attribution source/priority=] is equal to |b|'s [=attribution source/priority=] and |a|'s
+         [=attribution source/source time=] is less than |b|'s [=attribution source/source time=].
+1. Let |sourceToAttribute| be the first item in |matchingSources|.
+1. If the result of running
+    [=match an attribution source's filter data against filters=] with |sourceToAttribute| and |trigger|'s [=attribution trigger/filters=] is false, return.
+1. If the result of running [=should attribution be blocked by rate limit=] with |trigger| and
+    |sourceToAttribute| is <strong>blocked</strong>, return.
+1. Let |rateLimitRecord| be a new [=attribution rate-limit record=] with the items:
+    : [=attribution rate-limit record/scope=]
+    :: "`attribution`"
+    : [=attribution rate-limit record/source site=]
+    :: |sourceToAttribute|'s [=attribution source/source site=]
+    : [=attribution rate-limit record/attribution destination=]
+    :: |attributionDestination|
+    : [=attribution rate-limit record/reporting endpoint=]
+    :: |sourceToAttribute|'s [=attribution source/reporting endpoint=]
+    : [=attribution rate-limit record/time=]
+    :: |trigger|'s [=attribution trigger/trigger time=]
+1. If the result of running [=should processing be blocked by reporting-endpoint limit=] with
+    |rateLimitRecord| is <strong>blocked</strong>, return.
+1. Let |eventLevelResult| be the result of running [=trigger event-level attribution=].
+1. Let |aggregatableResult| be the result of running [=trigger aggregatable attribution=].
+1. If both |eventLevelResult| and |aggregatableResult| are <strong>dropped</strong>, return.
```

Comment:
Could you help me understand what the behavior difference is here?

Reply:
I don't think there is behavioral change for event-level reports except that the rate-limit checks are now performed earlier (compared to the implementation).

Basically we do all the common steps first, and then try to create event-level and aggregatable reports separately, delete all other matched sources and update rate limits if at least one of them succeeds.

Reply:
And the difference between dropped, cache full and attributed is:
If dropped, then do nothing.
if cache full, all other matching sources should be deleted, but rate limits should not be updated.
If attributed, all other matching sources should be deleted and rate limits should be updated.